Snowplow/Internal/Session/SPSession.m

```diff
@@ -145,6 +145,7 @@ - (NSDictionary *) getSessionDictWithEventId:(NSString *)eventId eventTimestamp:
     if (userAnonymisation) { // mask the user identifier
         NSMutableDictionary *copy = [[NSMutableDictionary alloc] initWithDictionary:context];
         [copy setValue:kSPSessionAnonymousUserId forKey:kSPSessionUserId];
+        [copy setValue:[NSNull null] forKey:kSPSessionPreviousId];
```

Comment:
Question: I wonder if this would be the desired behaviour. I see the reason behind removing the previous sessionId as it would let to expose the old userId. Not sure but even in the current implementation we could get the old userId checking the events of the current session pre-anonymisation because it seems we don't reset the session when we anonymise. If we reset the session when anonymised removing the first SessionPreviousId, we could keep the session chain without exposing the session userId. Wdyt?

Reply:
Those are good points, but the reason why I wanted to anonymise the previous session ID was different. It was because it enables us in data modeling to connect all the anonymous sessions of a user together.

For example, if a user had the following sessions: S1 > S2 > S3. Since S3 has reference to S2 and S2 has reference to S1, we basically have the complete user ID as we can identify the whole usage of the app. I think this defeats the purpose of removing the userId from the session as it can be recreated in data modeling. The JavaScript tracker does not have the previous session reference so it doesn't happen there.

Your point about resetting the session ID when anonymous tracking is enabled or disabled is also valid. I don't know whether we should reset it automatically or let the user do it. The JavaScript tracker does not reset the session ID when anonymous tracking is changed and I think we should make the behaviour consistent with that. I actually wanted to change that in the JS tracker and [opened a PR some time ago](https://github.com/snowplow/snowplow-javascript-tracker/pull/1104#issuecomment-1238162512), but when discussing this, we decided it would be better to let the user reset the session when they want instead of forcing it since there are some use cases when they might not want it to happen.

Reply:
As we discussed, I changed the behaviour when `userAnonymisation` is toggled on the tracker controller – it now starts a new session. This should help with the problem that the same `sessionId` could be shared for non-anonymous and anonymous session when changing the setting.

I kept the anonymisation of the previous session ID because I think that still makes sense.